### ISSUE TITLE
Browser extension not detecting offscreen canvas passed to webworker?

### ISSUE THREAD

**Issue by kevzettler** [BabylonJS/Spector.js#149]
The browser extension doesn't seem to be detecting my a canvas that I have passed to a webworker

```
const offscreen = canvas.transferControlToOffscreen();
const worker = new Worker();
worker.postMessage(
  {
    type: "main/canvas-init",
    payload: {
      canvas:  offscreen,
      width: window.innerWidth,
      height: window.innerHeight,
      player: fetchLocalStoragePlayerData(),
    }
  },
  [offscreen]
);
```

Is this supported? Should I manually use the spector.js lib instead?

**Comment by sebavan** [comment]
Yup unfortunately it is not supported and even the lib would not be working due to the way we capture texture.

As this is getting more and more common, I really need to work on it but I won t have time before a month I guess.

**Comment by kevzettler** [comment]
Thanks @sebavan Any notes on the work needed? Maybe I or other community could hack on it.

**Comment by sebavan** [comment]
Feel free that would be perfect, I would say at first we would need to detect the offscreen and be able to record the calls to them by simply disabling the texture captures (who relies on canvas2D internally).

This would be a great start :-)

**Comment by XiaoDiDiDa** [comment]
is there any progress of this ticket?

**Comment by sebavan** [comment]
Not at the moment, would would like to give it a try ?

**Comment by Jeggery** [comment]
It has been three years and I really hope to see this feature implemented

**Comment by sebavan** [comment]
I wish I had more time, but I will not be able to work on it unfortunately. As it is open source, do you want to give it a try ?

**Comment by Jeggery** [comment]
> I wish I had more time, but I will not be able to work on it unfortunately. As it is open source, do you want to give it a try ?

This is a lot of difficulty for me, and I still need to delve deeper into the Spector source code and webgl principles

**Comment by junyuecao** [comment]
@sebavan  I created a PR which might this feature. 
https://github.com/BabylonJS/Spector.js/pull/298

@Jeggery Maybe you can have a try.

BTW, the extension is based on Chrome extension manifest versoin 2 which is deprecated and can't be update anymore without upgrading to version 3

**Comment by pmhpereira** [comment]
Hey @junyuecao.
I wanted to test your PR, but I haven't been able to compile your fork locally.
I ran `npm install` and `npm run build` and I'm getting a lot of typescript errors.

**Comment by junyuecao** [comment]
@pmhpereira 
Could you try with `npm run start`? It works on my Ubuntu 20.04

![image](https://github.com/BabylonJS/Spector.js/assets/1000408/6bf28cc7-e042-4792-9f09-6b6f6bc3fa77)

**Comment by sebavan** [comment]
I completely missed all of this !!! I am so sorry @junyuecao I will have a deep look ASAP but it sounds really good !!!

**Comment by junyuecao** [comment]
@sebavan 
I think the problem is that Google have deprecated the Chrome extesion manifest V2 so the extension can not be updated without updating to V3. I am not that familliar with Chrome extension so I didn't migrate to V3.

**Comment by Ctrlmonster** [comment]
Hey, I'm wondering what's the status on debugging offscreen canvases with Spector? 

 I thought #129 introduced offscreen canvas support? I'm trying the Chrome extension as well as the manual route via `spector.displayUI()` but the extension doesn't seem to find any offscreen canvases (regardless of the ticked option) and calling `spector.captureCanvas(canvasElement);` and `spector.spyCanvases();` both result in `No webgl context available on the chosen canvas. []`, the first immediately, the latter when when hitting record on the spawned UI.


I know this is open source so I don't want put any pressure :) – just wanted to ask for an update, as Offscreen Canvas becomes more popular (i.e. basically 100% support and widely used on webgame platforms like Poki) having a way to debug via Spector would be awesome 🙏

**Comment by sebavan** [comment]
Should be ok in latest master
